website/docs/function_utils.md

```diff
@@ -0,0 +1,72 @@
+---
+id: function_utils
+title: Function Utilities
+sidebar_label: Function Utilities
+---
+
+## Function Utility Functions
+
+These functions provide a unified way to retrieve or format a function's parameters
+or get the calling module of a function.
+
+### Functions and Usage
+
+- `build_value_by_func_parameters` this function accepts a `key` parameter in type `str` which contains the parameter's placeholders and then returns out the string by replacing the placeholders with the parameter's actual value.

```

Comment:
Try this. 
`build_value_by_func_parameters` format a new string by merging a unique message with the function parameters and custom parameter 


```Python
def build_value_by_func_parameters(key: str, func, *args, **kwargs) -> str:
    ....
```
`key`: base string for formatting the parameter. when not set the `func.__qualname__` is used
`func`: the function to extract it parameter.
` *args, **kwargs`: the function args/kwargs for building the result string